### PR TITLE
partial function for keys method

### PR DESCRIPTION
**Pull request by AlirezaSohofi**
# Description

This fixes a bug that causes PyTorch DataLoader to hang. `keys()` returns a generator, replacing it with a partial function would solve the issue.

Fixes # issue

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring including code style reformatting 
- [ ] Other (please describe):

# Checklist:

- [x] I have read the [contributing guideline doc](https://squirrel-core.readthedocs.io/en/latest/usage/code_of_conduct.html) (external contributors only)
- [ ] Lint and unit tests pass locally with my changes
- [x] I have kept the PR small so that it can be easily reviewed  
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] All dependency changes have been reflected in the pip requirement files.

### CODE REVIEW

File: squirrel/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = "0.15.0"
+__version__ = "0.16.0"
```
Comment (AlpAribal) [inline squirrel/__init__.py:1]: I think patch bump would also work here

**Review by AlpAribal — COMMENTED** [review]
LGTM after linting and tests pass.

**Comment by axkoenig** [comment]
Tried and tested this. It fixes the bug, thanks @AlirezaSohofi!

**Review by axkoenig — APPROVED** [review]

**Comment by pzdkn** [comment]
LGTM!

**Review by pzdkn — APPROVED** [review]
LGTM

**Review by axkoenig — APPROVED** [review]

**Pull request merged**